robot/src/main/java/frc/robot/commands/drivetrain/DrivetrainDefaultCommand.java

```diff
@@ -52,55 +53,39 @@ public void execute() {
             _xingTimer.reset();
         }
 
-        if (Robot.DRIVETRAIN_STATE == DrivetrainState.FREEHAND) {
-            if (_xingTimer.get() >= Constants.DRIVETRAIN_HOLD_POSITION_TIMER_THRESHOLD) {
-                Robot.DRIVETRAIN_SUBSYSTEM.holdPosition();
-            } else {
-                double cutPower = Robot.cutPower ? 0.5 : 1;
-                x = x * DrivetrainSubsystem.MAX_VELOCITY_METERS_PER_SECOND * cutPower;
-                y = y * DrivetrainSubsystem.MAX_VELOCITY_METERS_PER_SECOND * cutPower;
-                r = r * Constants.DRIVE_MAX_TURN_RADIANS_PER_SECOND * cutPower;
-
-                // angular position holder only acts if r == 0
-                r = AngularPositionHolder.GetInstance().getAngularVelocity(r, robotRotationDegrees.getRadians());
-
-                var targetChassisSpeeds = ChassisSpeeds.fromFieldRelativeSpeeds(
-                    x, // x translation
-                    y, // y translation
-                    r, // rotation
-                    robotRotationDegrees // The angle of the robot as measured by a gyroscope.
-                );
-
-                targetChassisSpeeds.vxMetersPerSecond = Slew.GetSlewedTarget(_velocityXSlewRate, targetChassisSpeeds.vxMetersPerSecond, _chassisSpeeds.vxMetersPerSecond);
-                targetChassisSpeeds.vyMetersPerSecond = Slew.GetSlewedTarget(_velocityYSlewRate, targetChassisSpeeds.vyMetersPerSecond, _chassisSpeeds.vyMetersPerSecond);
-                targetChassisSpeeds.omegaRadiansPerSecond = Slew.GetSlewedTarget(_angularSlewRate, targetChassisSpeeds.omegaRadiansPerSecond, _chassisSpeeds.omegaRadiansPerSecond);
-                _chassisSpeeds = targetChassisSpeeds;
-
-                Robot.DRIVETRAIN_SUBSYSTEM.drive(targetChassisSpeeds);
-            }
-        }
-        else if (Robot.DRIVETRAIN_STATE == DrivetrainState.ROBOT_ALIGN) {
+        
+        if (_xingTimer.get() >= Constants.DRIVETRAIN_HOLD_POSITION_TIMER_THRESHOLD) {
+            Robot.DRIVETRAIN_SUBSYSTEM.holdPosition();
+        } else {
             double cutPower = Robot.cutPower ? 0.5 : 1;
             x = x * DrivetrainSubsystem.MAX_VELOCITY_METERS_PER_SECOND * cutPower;
             y = y * DrivetrainSubsystem.MAX_VELOCITY_METERS_PER_SECOND * cutPower;
-
-            r = getAngularVelocityForAlignmentFromRadiansOffset(Robot.DRIVETRAIN_SUBSYSTEM.getTargetRadiansForCenterOfSpeakerTx());
+            r = r * Constants.DRIVE_MAX_TURN_RADIANS_PER_SECOND * cutPower;
+
+            // From the docs it sounds like this boolean will ONLY be true if it sees
+            // the priority tag, but we will have to double check to be certain.
+            if (Robot.DRIVETRAIN_STATE == DrivetrainState.ROBOT_ALIGN && Robot.LIMELIGHT_BACK.hasVisionTargetBoolean()) {
+                var txDegrees = Robot.LIMELIGHT_BACK.getBufferedTx();
+                var targetAngleDegrees = robotRotation.getDegrees() - txDegrees;
+                var targetAngleRadians = Math.toRadians(targetAngleDegrees);
```

Comment:
I still think we should try for buffering `targetAngleRadians`, as that will allow turning towards the target effectively even when we lose vision of the target.

Reply:
I think that could be an improvement but I don't think we have time to deal with it before Troy; it will require too much testing time

Reply:
How would that be any more difficult than testing the use of a buffered Tv?